documentation/api/package-dependency-management.md

```diff
@@ -0,0 +1,131 @@
+# Package Dependency Management
+
+We use [Poetry](https://python-poetry.org/docs/) for managing our APIs dependencies.
+
+The [pyproject.toml](../../api/pyproject.toml) file is where we configure the minimum or exact version of a dependency. These versions also get modified
+by the poetry commands (several described below) which can help generate / find the proper versions for a dependency.
+
+From this configuration, a [poetry.lock](../../api/poetry.lock) is generated which has the specific references to dependencies to be downloaded. The lock
+file should never be manually modified and should always be generated by Poetry.
+
+# General Information
+
+## Dependency Versions
+
+In our dependencies we could specify a package version as `1.2.0` or `^1.2.0` which each have specific semantic meaning (in this case, exactly `1.2.0` or `>=1.2.0 and <2.0.0`).
+
+See [version constraints](https://python-poetry.org/docs/dependency-specification/#version-constraints) for further details.
+
+## Package Extras
+
+Some dependencies have "extra" packages that can be downloaded in addition to the main package for additional optional features.
+These can be specified in the `pyproject.toml` file like so: `marshmallow-dataclass = {extras = ["enum", "union"], version = "^8.5.8"}` where
+each "extra" is a string in the extras list.
+
+## Dev Dependencies
+
+We split dependencies into dev and non-dev dependencies. Dev dependencies are ONLY installed locally and in some CI/CD commands as they are only necessary for local development (including tests, linting, formatting, and dev utilities).
+
+Many of our dev tools are used for testing, including generating data which we want to explicitly avoid doing non-locally.
+Doing this also reduces the image size, and improves security as there are fewer dependencies that could potentially run on a production server.
+
+Dependencies are grouped under `tool.poetry.dependencies` and `tool.poetry.group.dev.dependencies`. When adding a dependency,
+make sure to put it in the proper group as when the API runs non-locally it will only have  packages from the first group.
+
+# Common Commands
+
+See the Poetry docs for more details, but here are a few useful commands for managing package dependencies in common scenarios.
+
+## Installing
+
+Installing is only necessary if you're running outside of Docker (eg. you've set the `PY_RUN_APPROACH=local` env variable)
+
+```shell
+poetry install --no-root --all-extras --with dev
```

Comment:
Should we update `make setup-local` to match this and recommend using it instead?

https://github.com/HHS/simpler-grants-gov/blob/b6a4f32052d4e5ef828e98a8a0a87aa052f85d92/api/Makefile#L79-L88